src/games/strategy/triplea/delegate/StrategicBombingRaidBattle.java

```diff
@@ -203,7 +203,19 @@ public void fight(final IDelegateBridge bridge) {
     showBattle(bridge);
     final List<IExecutable> steps = new ArrayList<>();
     if (hasAA) {
-      steps.add(new FireAA());
+      boolean foundSpecificAA = false;
```

Comment:
You can tighten this block up with:
```    
if (hasAA) {
      // global1940 rules - each target type fires an AA shot against the planes bombing it
      m_targets.entrySet().stream()
          .filter(entry -> entry.getKey().getUnitAttachment().getIsAAforBombingThisUnitOnly())
          .forEach(entry -> steps.add(new FireAA(entry.getValue())));

      // otherwise fire an AA shot at all the planes
      if(steps.isEmpty()) {
        steps.add(new FireAA());
      }
    }
```
The above is NPE safe without the NPE checks, and avoids the tracking variable `foundSpecificAA`